crypto/ex_data.c

```diff
@@ -287,8 +287,6 @@ int CRYPTO_dup_ex_data(int class_index, CRYPTO_EX_DATA *to,
         CRYPTOerr(CRYPTO_F_CRYPTO_DUP_EX_DATA, ERR_R_MALLOC_FAILURE);
         return 0;
     }
-    if (!CRYPTO_set_ex_data(to, mx - 1, NULL))
-        goto err;
```

Comment:
That was to simplify the error handling in the loop below.
If the CRYPTO_set_ex_data succeeds the underlying
stack holds enough elements so that the CRYPTO_set_ex_data
in the loop below is guaranteed to succeed.

What problems do you see with that code?